sys/behavedefs/src/BTShow.hs

```diff
@@ -20,126 +20,109 @@ where
+         ++ "( " ++ pshow bnode2 ++ " )"
     pshow (BNdisable bnode1 bnode2)
-      =  "( " ++ (pshow bnode1) ++ " )\n"
-         ++ "[>>\n" ++ "( " ++ (pshow bnode2) ++ " )"
+      =  "( " ++ pshow bnode1 ++ " )\n"
+         ++ "[>>\n" ++ "( " ++ pshow bnode2 ++ " )"
     pshow (BNinterrupt bnode1 bnode2)
-      =  "( " ++ (pshow bnode1) ++ " )\n" ++ "[><\n"
-         ++ "( " ++ (pshow bnode2) ++ " )"
+      =  "( " ++ pshow bnode1 ++ " )\n" ++ "[><\n"
+         ++ "( " ++ pshow bnode2 ++ " )"
     pshow (BNhide chans bnode)
       =  "HIDE "
-         ++ ( Utils.join "; " [ n ++ " :: " ++ ( Utils.join " # " (map pshow srts) )
-                              | ChanId n _uid srts <- chans
-                              ] ) ++ " IN\n"
-         ++ (pshow bnode) ++ "\n"
+         ++ Utils.join "; " [ n ++ " :: " ++ Utils.join " # " (map pshow srts)
+                            | ChanId n _uid srts <- chans
+                            ]
+         ++ " IN\n"
+         ++ pshow bnode ++ "\n"
          ++ "NI\n"
 
-
 -- ----------------------------------------------------------------------------------------- --
 -- PShow INode
 
-
 instance PShow INode
   where
     pshow (BNbexpr (we,ivals) bexp)
       =  "VALENV "
-         ++ ( Utils.join ";\n"
-              (map (\(k,v) -> (pshow k) ++ " = " ++ (pshow v)) (Map.assocs ivals)) )
-         ++ ( Utils.join ";\n"
-              (map (\(k,v) -> (pshow k) ++ " = " ++ (pshow v)) (Map.assocs we)) )
-         ++ " IN " ++ (pshow bexp) ++ " NI"
+         ++ Utils.join ";\n" (map (\(k,v) -> pshow k ++ " = " ++ pshow v) (Map.assocs ivals))
+         ++ Utils.join ";\n" (map (\(k,v) -> pshow k ++ " = " ++ pshow v) (Map.assocs we))
+         ++ " IN " ++ pshow bexp ++ " NI"
     pshow (BNparallel chans snodes)
       =  case snodes of
-         { [] -> "STOP\n"
-         ; be -> "( " ++
-                 ( Utils.join (" )\n"++"|[ "++ (Utils.join ", " (map pshow chans)) ++" ]|\n( ")
-                              (map pshow be) ) ++
-                 " )"
-         }
+           [] -> "STOP\n"
+           be -> "( " 
+                 ++ Utils.join (" )\n"++"|[ "++ Utils.join ", " (map pshow chans) ++" ]|\n( ")
+                               (map pshow be)
+                 ++ " )"
     pshow (BNenable snode1 choffs snode2)
-      =  "( " ++ (pshow snode1) ++ " )\n" ++ ">>>"
+      =  "( " ++ pshow snode1 ++ " )\n" ++ ">>>"
          ++ case choffs of
-            { [] -> "\n"
-            ; _  -> " ACCEPT\n" ++ ( Utils.join ",\n" (map pshow choffs) ) ++ "\n" ++ "IN\n"
-            }
-         ++ "( " ++ (pshow snode2) ++ " )"
+              [] -> "\n"
+              _  -> " ACCEPT\n" ++ Utils.join ",\n" (map pshow choffs) ++ "\n" ++ "IN\n"
+         ++ "( " ++ pshow snode2 ++ " )"
     pshow (BNdisable snode1 snode2)
-      =  "( " ++ (pshow snode1) ++ " )\n"
-         ++ "[>>\n" ++ "( " ++ (pshow snode2) ++ " )"
+      =  "( " ++ pshow snode1 ++ " )\n"
```

Comment:
In the future we can consider using [`Turtle.Format`](https://hackage.haskell.org/package/turtle-1.4.2/docs/Turtle-Format.html).